holoviews/plotting/mpl/chart3d.py

```diff
@@ -109,45 +109,30 @@ class Scatter3DPlot(Plot3D, PointPlot):
                                      allow_None=True, doc="""
       Index of the dimension from which the sizes will the drawn.""")
 
-    def initialize_plot(self, ranges=None):
-        axis = self.handles['axis']
-        points = self.hmap.last
-        ranges = self.compute_ranges(self.hmap, self.keys[-1], ranges)
-        ranges = match_spec(points, ranges)
-        key = self.keys[-1]
-        xs, ys, zs = (points.dimension_values(i) for i in range(3))
-
-        style = self.style[self.cyclic_index]
-        cdim = points.get_dimension(self.color_index)
-        if cdim and 'cmap' in style:
-            cs = points.dimension_values(self.color_index)
-            style['c'] = cs
-            if 'clim' not in style:
-                clims = ranges[cdim.name]
-                style.update(vmin=clims[0], vmax=clims[1])
-        if points.get_dimension(self.size_index):
-            style['s'] = self._compute_size(points, style)
-
-        scatterplot = axis.scatter(xs, ys, zs, zorder=self.zorder, **style)
-
-        self.handles['axis'].add_collection(scatterplot)
-        self.handles['artist'] = scatterplot
-
-        return self._finalize_axis(key, ranges=ranges)
-
-    def update_handles(self, axis, points, key, ranges=None):
+    def get_data(self, element, ranges, style):
+        xs, ys, zs = (element.dimension_values(i) for i in range(3))
+        self._compute_styles(element, ranges, style)
+        # Temporary fix until color handling is deterministic in py3
```

Comment:
The behavior seems to be deterministic in python 2 but not python 3, not sure if it's down to dictionary ordering or something else.